experimental/host/netlink/onewire.go

```diff
@@ -247,35 +249,38 @@ func w1CmdWrite(d []byte) *w1Cmd { return &w1Cmd{typ: cmdWrite, payloadLen: len(
 
 //
 
-// socket is a netlink connector socket for reading and writing 1-wire messages.
-type socket struct {
-	fd int
+type socket interface {
```

Comment:
Note that this new interface is not needed right now, but will enable testing in a follow-up PR.